README.md

```diff
@@ -45,9 +47,81 @@ Optional:
   used for large support files like images or PDF documentation that would
   otherwise unnecessarily increase the bundle size. Don't exclude the README or
   the LICENSE.
+- `categories`: An array with up to three categories from the [list of
+  categories][categories] as strings to help users find the package.
 
+Packages always live in folders named as `{name}/{version}`. The name and
+version in the folder name and manifest must match. Paths in a package are local
+to that package. Absolute paths start in the package root while relative paths
+are relative to the file they are used in.
+
+### Templates
+Packages can act as templates for user projects. In addition to the module that
+a regular package provides, template packages also contain _starting points._ A
+starting point is a set of files that Typst copies into the directory of a new
+project. A package can provide multiple starting points. Packages are considered
+template packages if they contain at least one starting point.
+
+In most cases, the starting point files should not include the styling code for
+the template. Instead, the starting point's entrypoint file should import a
+function from the package. Then, use this function with a show rule to
+apply it to the rest of the document.
+
+Template packages (also informally called templates) can declare starting points
+in their `typst.toml` file. A template package's `typst.toml` could look like
+this:
+
+```toml
+[package]
+name = "charged-ieee"
+version = "0.1.0"
+entrypoint = "template.typ"
+authors = ["Typst GmbH <https://typst.app>"]
+license = "Unlicense"
```

Comment:
Not very important, but I think we should mention a different license here because the Unlicense is [not a good license](https://softwareengineering.stackexchange.com/questions/147111/what-is-wrong-with-the-unlicense).
Maybe use `MIT OR Apache-2.0` instead? Or `EUPLv1.2`, though that one isn't as well known. I must admit that there isn't really one obvious choice for packages and templates.

Reply:
typst is already Apache-2.0, so that works?

Reply:
Switched to MIT for package and MIT-0 for templates. See the updated README for more details.

Reply:
Sounds good to me